src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -467,4 +484,82 @@ public List<Project> getProjects(Vulnerability vulnerability) {
         projects.addAll(set);
         return projects;
     }
+
+    public synchronized VulnerabilityAlias synchronizeVulnerabilityAlias(VulnerabilityAlias alias) {
+        final Map<String, Object> params = new HashMap<>();
+        String filter = "";
+        if (alias.getCveId() != null) {
+            filter += "(cveId == :cveId)";
+            params.put("cveId", alias.getCveId());
+        }
+        if (alias.getSonatypeId() != null) {
+            if (filter != null) filter += " && ";
```

Comment:
This will always be false as filter is set as an empty string. 
Need to change it to `if (!filter.isEmpty())`